uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/leave-request-popup-controller.js

```diff
@@ -545,13 +625,166 @@ define([
+        return vm.leaveRequest.roleOf({
+            id: managerContactId
+          })
+          .then(function (role) {
+            if (role === 'manager') {
+              vm.uiOptions.isManager = true;
+            } else {
+              vm.uiOptions.isManager = false;
+            }
+          });
+      }
+
+      /**
+       * initialize absence types
+       *
+       */
+      function initAbsenceType() {
+        if (canEdit()) {
+          vm.selectedAbsenceType = _.find(vm.absenceTypes, function (absenceType) {
+            return absenceType.id == vm.leaveRequest.type_id;
+          });
+        } else {
+          // Assign the first absence type to the leave request
+          vm.selectedAbsenceType = vm.absenceTypes[0];
+        }
+
+        vm.leaveRequest.type_id = vm.selectedAbsenceType.id;
+        // Init the `balance` object based on the first absence type
+        vm.balance.opening = vm.selectedAbsenceType.remainder;
+      }
+
+      /**
+       * initialize from and to dates, day types and contact
+       *
+       */
+      function initDates() {
+        initStatus();
+        initDayTypes();
+        initContact();
+
+        if (canEdit()) {
+          vm.uiOptions.fromDate = convertDateFormatFromServer(vm.leaveRequest.from_date);
+          vm.onDateChange(vm.uiOptions.fromDate, 'from');
+          vm.uiOptions.toDate = convertDateFormatFromServer(vm.leaveRequest.to_date);
+          vm.onDateChange(vm.uiOptions.toDate, 'to');
+        }
+      }
+
+      /**
+       * initialize day types
+       *
+       */
+      function initDayTypes() {
+        if (canEdit()) {
+          vm.uiOptions.selectedFromType = getDateTypeFromValue(vm.leaveRequest.from_date_type);
+          vm.uiOptions.selectedToType = getDateTypeFromValue(vm.leaveRequest.to_date_type);
+        }
+      }
+
+      /**
+       * initialize status
+       *
+       */
+      function initStatus() {
+        if (canEdit()) {
+          vm.uiOptions.statusLabel = getStatusFromValue(vm.leaveRequest.status_id).label;
+          //waiting_approval is removed below so call the above before it
+          if (vm.uiOptions.isManager) {
+            setStatusesForManager();
+          }
+        } else {
+          vm.leaveRequest.status_id = valueOfRequestStatus('waiting_approval');
+        }
+      }
+
+      /**
+       * initialize contact
+       *
+       * {Promise}
+       */
+      function initContact() {
+        if (canEdit()) {         
+          return Contact.find(vm.leaveRequest.contact_id)
+            .then(function (contact) {
+              vm.uiOptions.contact = contact;
+            });
+        } else {
+          return $q.resolve({});
+        }
+      }
+
+      /**
+       * sets leave requestion statuses for manager
+       *
+       */
+      function setStatusesForManager() {
+        var allowed_statuses = ['approved', 'more_information_requested', 'cancelled'];
+        if (vm.uiOptions.isManager) {
```

Comment:
You are already checking that `isManager` is true before you are calling this function, plus the name of the function contains `ForManager`. You can skip this check